pkg/util/extract_forwarded_test.go

```diff
@@ -0,0 +1,59 @@
+package util
+
+import (
+	"context"
+	"testing"
+
+	"google.golang.org/grpc/metadata"
+)
+
+func TestGetSourceFromOutgoingCtx(t *testing.T) {
+	tests := []struct {
+		name  string
+		key   string
+		value string
+		want  string
+	}{
+		{
+			name:  "No value in key",
+			key:   ipAddressesKey,
+			value: "",
+			want:  "",
+		},
+		{
+			name:  "Value in key",
+			key:   ipAddressesKey,
+			value: "172.16.1.1",
+			want:  "172.16.1.1",
+		},
+		{
+			name:  "Stored under wrong key",
+			key:   "wrongkey",
+			value: "172.16.1.1",
+			want:  "",
```

Comment:
If you define your custom data `type`, this test can be removed. You shouldn't really worry about the "wrong key".

Reply:
As mentioned before GRPC metadata doesn't support custom types.